ui/conductor/src/widgets/graphic/GraphicLayer.vue

```diff
@@ -0,0 +1,221 @@
+<template>
+  <!-- eslint-disable vue/no-v-html -->
+  <div
+      v-html="svgRaw"
+      class="svg--container"
+      @click="onSvgClick"
+      ref="svgContainerEl"/>
+</template>
+
+<script setup>
+import {closeResource} from '@/api/resource.js';
+import {usePullTrait} from '@/traits/traits.js';
+import {loadConfig} from '@/widgets/graphic/config.js';
+import {usePathUtils} from '@/widgets/graphic/path.js';
+import {useSvgEffects} from '@/widgets/graphic/svg.js';
+import {effectScope, nextTick, onUnmounted, reactive, ref, watch} from 'vue';
+
+const props = defineProps({
+  layer: {
+    type: Object,
+    required: true
+  },
+  // Either null (for no selection), the index of the element that is selected, or an array of selected element indexes
+  selected: {
+    type: [Number, Array],
+    default: null
+  },
+  // control whether one or multiple elements can be selected
+  multiple: {
+    type: Boolean,
+    default: false
+  }
+});
+const emit = defineEmits(['click:element', 'update:selected']);
+
+const _selected = ref(null);
+watch(() => props.selected, (v) => _selected.value = v);
+
+/**
+ * @param {PointerEvent} event
+ */
+const onSvgClick = (event) => {
+  const el = event.target.closest('[data-element-idx]');
+  if (!el) return; // not an interesting click
+  const elementIdx = parseInt(el.dataset.elementIdx);
+  const layerElement = config.value.elements[elementIdx];
+  emit('click:element', {element: layerElement, index: elementIdx, event});
+  selectElement(el, elementIdx, layerElement);
+};
+
+const selectElement = (el, elementIdx, element) => {
```

Comment:
Unused parameters here,  might be used in the future though?

Reply:
Good spot, you're right they are unused and there just in case.